### DISCUSSION TITLE
Example on how to use with LaTeX

### DISCUSSION THREAD

**Discussion by kpeeters** [bwoods/TeX--#7] [General]
Do you have a simple example on how you intend people to use TeX-- with LaTeX? Is the idea to overload the `tex` class similar to what you did with `plain`?

**Comment by bwoods** [comment]
> Is the idea to overload the `tex` class similar to what you did with `plain`?

Yes. 

You will need to ensure that all of your LaTeX macro files are in your `tex` folder (or are found in your `search_dir`).

> Any additional macro files needed by the application should be stored in the `tex` folder. Although, if they are in the `search_dir` directly, they will be found.

Afterward, it should as simple as preloading any packages your documents assume are already loaded, similar to what  `tex::plain`’s constructor does with`R"(\input plain)"`:

```cpp
virtual void typeset(const std::string& filename, std::streambuf * resultbuf, const std::string& search_dir, const std::string& working_dir, std::ostream& output)
{
	input_path = search_dir;
	if (input_path.empty() == false and input_path.back() != '/')
		input_path.push_back('/');

	output_path = working_dir;
	if (output_path.empty() == false and output_path.back() != '/')
		output_path.push_back('/');

	input_stream = new std::stringstream; // will be closed as term_in
	output_stream = new std::iostream(output.rdbuf()); // will be closed as term_out

	if (resultbuf) {
		dvi_file = new std::stringstream;
		dvi_file->rdbuf(resultbuf);
	}

	tex::typeset({
		R"(\nonstopmode)", // omits all stops (\batchmode also omits terminal output)
		R"(\input plain)",
		R"(\input)", filename.c_str(),
		R"(\end)",
	});
}
```
